x/concentrated-liquidity/position_test.go

```diff
@@ -496,3 +496,257 @@ func (s *KeeperTestSuite) TestCalculateUnderlyingAssetsFromPosition() {
+			expectedNewPositionId:   0,
+			expectedErr:             types.PositionOwnerMismatchError{PositionOwner: secondAddress.String(), Sender: defaultAddress.String()},
+		},
+		{
+			name: "Error: Fungify three positions, but one of them is not in the same range",
+			setupFullyChargedPositions: []position{
+				{1, 1, defaultAddress, DefaultCoin0, DefaultCoin1, DefaultLowerTick, DefaultUpperTick},
+				{2, 1, defaultAddress, DefaultCoin0, DefaultCoin1, DefaultLowerTick - 1, DefaultUpperTick},
+				{3, 1, defaultAddress, DefaultCoin0, DefaultCoin1, DefaultLowerTick, DefaultUpperTick},
+			},
+			positionIdsToMigrate:    []uint64{1, 2, 3},
+			accountCallingMigration: defaultAddress,
+			expectedNewPositionId:   0,
+			expectedErr:             types.PositionsNotInSameTickRangeError{Position1TickLower: DefaultLowerTick - 1, Position1TickUpper: DefaultUpperTick, Position2TickLower: DefaultLowerTick, Position2TickUpper: DefaultUpperTick},
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			// Init suite for each test.
+			s.Setup()
+			s.Ctx = s.Ctx.WithBlockTime(defaultBlockTime)
+			totalPositionsToCreate := sdk.NewInt(int64(len(test.setupFullyChargedPositions) + len(test.setupUnchargedPositions)))
+			requiredBalances := sdk.NewCoins(sdk.NewCoin(ETH, DefaultAmt0.Mul(totalPositionsToCreate)), sdk.NewCoin(USDC, DefaultAmt1.Mul(totalPositionsToCreate)))
+
+			// Fund accounts
+			s.FundAcc(defaultAddress, requiredBalances)
+			s.FundAcc(secondAddress, requiredBalances)
+
+			// Create two default CL pools
+			s.PrepareConcentratedPool()
+			s.PrepareConcentratedPool()
+
+			// Set up fully charged positions
+			totalLiquidity := sdk.ZeroDec()
+			for _, pos := range test.setupFullyChargedPositions {
+				_, _, _, liquidityCreated, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, pos.poolId, pos.acc, pos.coin0.Amount, pos.coin1.Amount, sdk.ZeroInt(), sdk.ZeroInt(), pos.lowerTick, pos.upperTick)
+				s.Require().NoError(err)
+				totalLiquidity = totalLiquidity.Add(liquidityCreated)
+			}
+
+			// Increase block time by the fully charged duration
+			s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(types.FullyChargedDuration))
+
+			// Set up uncharged positions
+			for _, pos := range test.setupUnchargedPositions {
+				_, _, _, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, pos.poolId, pos.acc, pos.coin0.Amount, pos.coin1.Amount, sdk.ZeroInt(), sdk.ZeroInt(), pos.lowerTick, pos.upperTick)
+				s.Require().NoError(err)
+			}
+
+			// Increase block time by one more day
+			s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(time.Hour * 24))
+
+			newPositionId, err := s.App.ConcentratedLiquidityKeeper.FungifyChargedPosition(s.Ctx, test.accountCallingMigration, test.positionIdsToMigrate)
+			if test.expectedErr != nil {
+				s.Require().Error(err)
+				s.Require().ErrorIs(err, test.expectedErr)
+				s.Require().Equal(test.expectedNewPositionId, newPositionId)
```

Comment:
should this always be
```suggestion
				s.Require().Equal(0, newPositionId)
```